clusterloader2/pkg/measurement/common/kube_proxy_iptables_restore_failures.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"fmt"
+	"time"
+
+	"k8s.io/klog/v2"
+	"k8s.io/perf-tests/clusterloader2/pkg/errors"
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement"
+	"k8s.io/perf-tests/clusterloader2/pkg/util"
+)
+
+const (
+	kubeProxyIptablesFailuresName = "KubeProxyIptablesRestoreFailures"
+	// The metric definition
+	// https://github.com/kubernetes/kubernetes/blob/master/pkg/proxy/metrics/metrics.go#L129-L138
+	// IptablesPartialRestoreFailuresTotal is the number of iptables *partial* restore
+	// failures (resulting in a fall back to a full restore) that the proxy has seen.
+	// This should not happen unless there is a kube-proxy error on the code
+	// Ref: https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/3453-minimize-iptables-restore
+	kubeProxyIptablesFailuresQuery = `sum(kubeproxy_sync_proxy_rules_iptables_partial_restore_failures_total)`
+)
+
+func init() {
+	create := func() measurement.Measurement {
+		return CreatePrometheusMeasurement(&kubeProxyIptablesFailuresGatherer{})
+	}
+	if err := measurement.Register(kubeProxyIptablesFailuresName, create); err != nil {
+		klog.Fatalf("Cannot register %s: %v", kubeProxyIptablesFailuresName, err)
+	}
+}
+
+type kubeProxyIptablesFailuresGatherer struct{}
+
+type iptablesRestoreFailures struct {
+	Total int `json:"total"`
+}
+
+func (c *kubeProxyIptablesFailuresGatherer) Gather(executor QueryExecutor, startTime, endTime time.Time, config *measurement.Config) ([]measurement.Summary, error) {
+	iptablesFailures, err := c.gatherKproxyIptablesFailures(executor, startTime, endTime)
+	if err != nil {
+		return nil, err
+	}
+
+	content, err := util.PrettyPrintJSON(iptablesFailures)
+	if err != nil {
+		return nil, err
+	}
+
+	summaries := []measurement.Summary{measurement.CreateSummary(kubeProxyIptablesFailuresName, "json", content)}
+	if iptablesFailures.Total > 0 {
+		return summaries, errors.NewMetricViolationError("kubeproxy_sync_proxy_rules_iptables_partial_restore_failures_total", fmt.Sprintf("iptables restore failures greater than 0, got: %d", iptablesFailures.Total))
+	}
+	return summaries, nil
+}
+
+func (c *kubeProxyIptablesFailuresGatherer) gatherKproxyIptablesFailures(executor QueryExecutor, startTime, endTime time.Time) (*iptablesRestoreFailures, error) {
+	measurementEnd := time.Now()
```

Comment:
It seems that you are using time.Now() as endTime, instead of endTime passed via argument. This probably breaks tests as data loaded by lazyloader starts at time.Unix(0, 0). I can see that in test you have properly passed start and end time, but you are not respecting it here

Reply:
copy paste thing

Reply:
same error :/ after removing this line